tools/update-packagefiles.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+
+# Copyright 2021 Xavier Claessens <xclaesse@gmail.com>
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import subprocess
+import configparser
+import tarfile
+import shutil
+from pathlib import Path
+
+def read_wrap(filename):
+    wrap = configparser.ConfigParser(interpolation=None)
+    wrap.read(filename)
+    return wrap[wrap.sections()[0]]
+
+if __name__ == '__main__':
+    for f in Path('subprojects').glob('*.wrap'):
+        wrap_section = read_wrap(f)
+        patch_directory = wrap_section.get('patch_directory')
+        if not patch_directory:
+            continue
+        archive_path = Path('subprojects', 'packagecache', wrap_section['source_filename'])
+        if not archive_path.exists():
+            continue
+        archive = tarfile.open(archive_path)
```

Comment:
Also tarfile.open does not work for zip files, which is why in meson itself we handle unpacking via shutil.unpack_archive() -- which sadly does not support getting file lists.

I recently noticed this and wondered why we use zip files at all :D since they are rather larger on average, and at least since github archives supports both, we should prefer the nicer format.

Reply:
#167 because why not...

However, this won't help for upstream projects which host their own downloads and which use zip.

Reply:
Added support for zipfile now.